conda/resolve.py

```diff
@@ -380,24 +401,34 @@ def depends_on_(spec):
 
     def version_key(self, dist, vtype=None):
         rec = self.index[dist]
-        cpri = rec.get('priority', 1)
-        valid = 1 if cpri < MAX_CHANNEL_PRIORITY else 0
-        ver = normalized_version(rec.get('version', ''))
-        bld = rec.get('build_number', 0)
-        bs = rec.get('build')
+        channel = rec.channel
+        channel_priority = self._channel_priorities_map.get(channel.canonical_name, 1)
```

Comment:
I used `1` as the default value here, matching the previous default value.  I'm not so sure this shouldn't be set to something like MAX_CHANNEL_PRIORITY though.